padevents/padevents.py

```diff
@@ -666,50 +678,77 @@ async def _do_aed_add(self, ctx, item):
                 return
-            dmevents[index - 1]['offset'] = offset
+            dmevents[loc]['offset'] = offset
         await ctx.tick()
 
     @aed_e.command(name="searchstr")
-    async def aed_e_searchstr(self, ctx, index: int, *, searchstr):
+    async def aed_e_searchstr(self, ctx, index, *, searchstr):
         """Set search string of an autoeventdm"""
         searchstr = searchstr.strip('"')
+        loc = await self._get_and_validate_aed_index(ctx, index)
+        if loc is None:
+            return
         async with self.config.user(ctx.author).dmevents() as dmevents:
-            if not 0 < index <= len(dmevents):
-                await send_cancellation_message(ctx, "That isn't a valid index.")
+            if not await get_user_confirmation(ctx, "Modify the search string for {}. `{}` to `{}`?"
+                    .format(loc + 1, dmevents[loc]['searchstr'], searchstr)):
                 return
-            dmevents[index - 1]['searchstr'] = searchstr
+            dmevents[loc]['searchstr'] = searchstr
         await ctx.tick()
 
     @aed_e.command(name="toggle3p")
-    async def aed_e_toggle3p(self, ctx, index: int):
+    async def aed_e_toggle3p(self, ctx, index):
         """Include/exclude 3-player dungeons in an autoeventdm"""
+        loc = await self._get_and_validate_aed_index(ctx, index)
+        if loc is None:
+            return
         async with self.config.user(ctx.author).dmevents() as dmevents:
-            if not 0 < index <= len(dmevents):
-                await send_cancellation_message(ctx, "That isn't a valid index.")
-                return
-            event = dmevents[index - 1]['searchstr']
-            if dmevents[index - 1].get('include3p') is None:
+            event = dmevents[loc]['searchstr']
+            if dmevents[loc].get('include3p') is None:
                 # case of legacy configs
-                dmevents[index - 1]['include3p'] = True
-            if dmevents[index - 1]['include3p']:
-                dmevents[index - 1]['include3p'] = False
+                dmevents[loc]['include3p'] = True
+            if dmevents[loc]['include3p']:
+                dmevents[loc]['include3p'] = False
                 await send_confirmation_message(ctx, "I will **exclude** 3P dungeons for {}. `{}`"
-                                                .format(index, event))
+                                                .format(loc + 1, event))
                 return
-            dmevents[index - 1]['include3p'] = True
+            dmevents[loc]['include3p'] = True
             await send_confirmation_message(ctx, "I will **include** 3P dungeons for {}. `{}`"
-                                            .format(index, event))
+                                            .format(loc + 1, event))
             return
 
+    async def _get_and_validate_aed_index(self, ctx, user_index):
+        """Returns the location for the autoeventdm of interest, based on Python list indexing"""
```

Comment:
Since we're doing `corrected_index` instead of `location`, change the language in the docstring here too